### PR TITLE
Initial change to v1alpha4 types

### PR DESCRIPTION
**Pull request by prksu**
**What this PR does / why we need it**:

- Create types for v1alpha4
- Remove v1alpha2 types
- Create webhook conversion
- Upgrade golang to v1.16 (required by dependencies) [ref](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md#minimum-go-version)
- Upgrade controller-runtime to v0.9.0-alpha.1 (required by capi nighty release) [ref](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md#controller-runtime-version)
- Refactor manager to have a single manager watching all namespaces and answer to webhook calls [ref](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md#required-kustomize-changes-to-have-a-single-manager-watching-all-namespaces-and-answer-to-webhook-calls)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/issues/199

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
Add support cluster api v1alpha4 types 
```

/kind feature
/kind api-change

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Build the manager binary
-FROM golang:1.13.15 as builder
+FROM golang:1.16.2 as builder
```
Comment (cpanato) [inline Dockerfile:16]: nit: update to 1.16.3

File: Makefile [hunk 2]
```diff
@@ -348,7 +347,7 @@ release-binary: $(RELEASE_DIR)
 		-e GOARCH=$(GOARCH) \
 		-v "$$(pwd):/workspace" \
 		-w /workspace \
-		golang:1.13.15 \
+		golang:1.16.2 \
```
Comment (cpanato) [inline Makefile:350]: update to 1.16.3

**Review by cpanato — COMMENTED** [review]
overal lgtm
just some non blocking nits

File: Makefile (hunk 2 above)
Comment (prksu) [reply Makefile]: addressed

File: Dockerfile (hunk 1 above)
Comment (prksu) [reply Dockerfile]: addressed

**Comment by prksu** [comment]
@cpanato ptal, thanks!

**Review by cpanato — APPROVED** [review]
/lgtm

thanks for working on this!

**Pull request merged**